### PR TITLE
ARM64: Enable Varargs with float argument.

### LINKED ISSUE #2989: ARM64: Assertion failed 'assert: (allRegs(theInterval->registerType) & mask) != 0
This is caused when we have floating point arguments passed to vararg methods
There also needs to be a fix to properly home floating point arguments in a varargs method.

### PR DESCRIPTION
**Pull request by kyulee1**
Fixes #2989
ARM64 ABI does not pass float register as arguments for vararg calls.
All such float arguments should be copied to integer registers.
In Lower, I explicitly added a GT_COPY for such case and support code-gen
later to expand it properly using `fmov`.
I also skip genFnPrologCalleeRegArgs() for vararg since we've already
saved all integer registers in the prolog.

### CODE REVIEW

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64 Checked
@dotnet-bot test Windows_NT arm64 Release

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64 Checked
@dotnet-bot test Windows_NT arm64 Release

**Comment by kyulee1** [comment]
@dotnet/jit-contrib @dotnet/arm64-contrib PTAL

File: src/jit/lower.cpp [hunk 1]
```diff
@@ -1499,6 +1499,25 @@ void Lowering::LowerArg(GenTreeCall* call, GenTreePtr* ppArg)
         else
 #endif // !defined(_TARGET_64BIT_)
         {
+
+#ifdef _TARGET_ARM64_
+            // For vararg call, reg args should be all integer.
+            // Insert a copy to reinterpret float as integer.
```
Comment (CarolEidt) [inline src/jit/lower.cpp]: Minor suggestion: I would say "Insert a copy to move float value to integer register."

**Comment by CarolEidt** [comment]
LGTM

File: src/jit/lower.cpp (hunk 1 above)
Comment (kyulee1) [inline src/jit/lower.cpp]: Thanks. Will update the comment on merge.

**Comment by briansull** [comment]
LGTM

**Pull request merged**